lib/engine/game/g_1858/game.rb

```diff
@@ -283,6 +288,25 @@ def submit_revenue_str(routes, _show_subsidy)
             "#{format_revenue_currency(private_revenue)} private revenue"
         end
 
+        def buy_train(operator, train, price = nil)
+          bought_from_depot = (train.owner == @depot)
+          super
+          return if @phase7_trains_bought >= @phase7_train_trigger
```

Comment:
wouldn't be easier/more robust to check that the train already rusted?

Reply:
I'd hope that this code doesn't skip any train buys, but it could be changed. Is there an easy way of checking if a class of train has rusted?